ddl/ddl_api.go

```diff
@@ -244,10 +244,15 @@ func buildColumnAndConstraint(ctx sessionctx.Context, offset int,
 
 // checkColumnCantHaveDefaultValue checks the column can have value as default or not.
 // Now, TEXT/BLOB/JSON can't have not null value as default.
-func checkColumnCantHaveDefaultValue(col *table.Column, value interface{}) (err error) {
+func checkColumnCantHaveDefaultValue(ctx sessionctx.Context, col *table.Column, value interface{}) (err error) {
```

Comment:
how about this?
```go
func checkColumnCantHaveDefaultValue(ctx sessionctx.Context, col *table.Column, value interface{}) (error, bool) {
	if value != nil && (col.Tp == mysql.TypeJSON ||
		col.Tp == mysql.TypeTinyBlob || col.Tp == mysql.TypeMediumBlob ||
		col.Tp == mysql.TypeLongBlob || col.Tp == mysql.TypeBlob) {
		if !ctx.GetSessionVars().SQLMode.HasStrictMode() && value == "" {
			sc := ctx.GetSessionVars().StmtCtx
			sc.AppendWarning(errBlobCantHaveDefault.GenByArgs(col.Name.O))
			return nil, false
		}
		// TEXT/BLOB/JSON can't have not null default values.
		return errBlobCantHaveDefault.GenByArgs(col.Name.O), false
	}
	return nil, true
}
```
and in line 322:
```go
				if err , hasDefaultValue = checkColumnCantHaveDefaultValue(ctx,col, value); err != nil {
					return nil, nil, errors.Trace(err)
				}
				if hasDefaultValue {
					col.DefaultValue = value
					removeOnUpdateNowFlag(col)
				}
```
and in line 1443
```go
			if err ,hasDefaultValue = checkColumnCantHaveDefaultValue(ctx, col, value); err != nil {
				return errors.Trace(err)
			}
			if hasDefaultValue {
				col.DefaultValue = value
			}
```
then we can remove `ignoreNullDefaultValuesForColumns ` function